fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/ImplementDistinctUnionRule.java

```diff
@@ -175,41 +173,14 @@ public void onMatch(@Nonnull final CascadesRuleCall call) {
                     if (merge.isEmpty()) {
                         merge.add(Pair.of(orderings.get(0), orderings.get(0)));
                     } else {
-                        //
-                        //TODO We currently are unable to incrementally compute a merged order as the operation is
-                        //  not associative in all cases. The reason for that is that orderings are not represented
-                        //  as PartialOrder<KeyPart> but as List<KeyPart>. While we do the reasoning for all
-                        //  ordering-related logic in PartialOrder space, we transform back to a list which proves
-                        //  to be lossy in some cases causing associativity to not hold).
-                        //  https://github.com/FoundationDB/fdb-record-layer/issues/1618
-                        
-                        //TODO
-                        //  final var lastMerged = merge.size() - 1;
-                        //  final Ordering mergedOrdering =
-                        //      OrderingVisitor.deriveForUnionFromOrderings(ImmutableList.of(merge.get(lastMerged).getKey(), orderings.get(merge.size())),
-                        //          RequestedOrdering.preserve(),
-                        //          Ordering::intersectEqualityBoundKeys);
-                        //
-                        final var currentOrderings = ImmutableList.<Ordering>builder()
-                                .addAll(merge.stream().map(Pair::getValue).collect(ImmutableList.toImmutableList()))
-                                .add(orderings.get(merge.size()))
-                                .build();
+                        final var lastMerged = merge.size() - 1;
```

Comment:
Maybe we can discuss this offline to make sure I understand the logic here correctly.